### PR TITLE
Constructable Chem Master

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Ports over the constructable chemmaster from TG.

Also fixes up a number of colon usage in the file....seriously, why was it done this way?

### CODE REVIEW

**Comment by TheDZD** [comment]
Any functional differences between this and the normal Chemmaster?

**Comment by Crazylemon64** [comment]
I think, that it contains parts, and can be (de/re)constructed. Looks like part upgrades do nothing for it as-is, though perhaps an added matter bin to control how much you can keep in the buffer, and a manipulator enabling the creation of more advanced (larger) patches and pills might be nice additions. Or maybe I have no idea of balance?

**Comment by Fox-McCloud** [comment]
>Any functional differences between this and the normal Chemmaster?

You can deconstruct it--aside from that, no functional difference.

File: code/modules/reagents/Chemistry-Machinery.dm [hunk 1]
```diff
@@ -759,6 +759,72 @@
...
+
+	if(panel_open)
+		if(istype(B, /obj/item/weapon/crowbar))
+			default_deconstruction_crowbar(B)
+			return 1
+		else
+			user << "<span class='warning'>You can't use the [src.name] while it's panel is opened!</span>"
+			return 1
+
+	if(istype(B, /obj/item/weapon/reagent_containers/glass) || istype(B, /obj/item/weapon/reagent_containers/food/drinks/drinkingglass))
```
Comment (Crazylemon64) [inline code/modules/reagents/Chemistry-Machinery.dm]: Perhaps I'm a ninny, but why not simply a parent call here?

Comment (Fox-McCloud) [inline code/modules/reagents/Chemistry-Machinery.dm]: that would probably be ok; just didn't think about it, TBH.

**Comment by FalseIncarnate** [comment]
If there is no difference, perhaps we should just flat out replace the existing ones with the build-able versions, much like we did with the kitchen update and ovens/grills/etc.

Unless there is some sort of difference, it just doesn't make sense to keep a functionally identical one to confuse potential mappers.

**Comment by Fox-McCloud** [comment]
I'd rather not do that in the event that upgrades to it are designed, in the future and we want to differentiate more between the non-constructable and constructable ones (like the dispensers).

**Comment by Plotron** [comment]
I agree with @Crazylemon64, we could apply some fancy buffer adjustments.
And maybe for the top-tier upgrades make it possible to make 70-unit pills or something.

**Pull request merged**